lte/gateway/c/core/oai/tasks/nas/emm/msg/AttachAccept.cpp

```diff
@@ -99,7 +99,7 @@ int decode_attach_accept(attach_accept_msg* attach_accept, uint8_t* buffer,
       case ATTACH_ACCEPT_LOCATION_AREA_IDENTIFICATION_IEI:
         if ((decoded_result = decode_location_area_identification_ie(
                  &attach_accept->locationareaidentification,
-                 ATTACH_ACCEPT_LOCATION_AREA_IDENTIFICATION_IEI,
+                 ATTACH_ACCEPT_LOCATION_AREA_IDENTIFICATION_IEI != 0,
```

Comment:
I am not sure if I understand correctly (I am not that fluent in the mme business logic) - but, is this related to the reduction of warnings? If not, I would keep it out of scope here.

Reply:
In my opinion, let's retain your changes because earlier also the condition always evaluates to true. On failing to receive expected IE, the macro "CHECK_IEI_DECODER" will take care to return error, "TLV_UNEXPECTED_IEI"